### ISSUE TITLE
Scrolling the list of trackers is nonobvious on mac computers

### ISSUE THREAD

**Issue by Zakkai** [EFForg/privacybadgerchrome#591]
![screen shot 2015-08-28 at 10 38 14 pm](https://cloud.githubusercontent.com/assets/1514445/9560190/d056653c-4dd5-11e5-86d5-9361e30c8d67.png)

On Google docs, where there are lots of trackers listed by the Badger, if mouse over the bottom listing, the pop-up tip is obscured under the "Disable Privacy Badger for This Site" button.

I'm in Chrome, running OSX Yosemite 10.10.1

**Comment by cooperq** [comment]
You can scroll the list of domains down. then you should see the popup for the bottom listing.

**Comment by Zakkai** [comment]
That's true, but in Chrome, the scroll bar doesn't display by default, until you begin to scroll. This means that there is no visual queue that scrolling is what you need to do, and I, a relatively savvy user, didn't guess it without the visual queue. I think it'd be great if there was some indication that scrolling was necessary. Perhaps a permanently visible scroll bar, or a border around the scrollable region.

Just my suggestion -- I know you have to prioritize more serious problems most of the time.

**Comment by cooperq** [comment]
Yea that seems to be the case with OSX that they hide the scrollbar until you use it. Which is bad UI IMO. We should think about how to indicate to a user that they can scroll down.

**Comment by karisafari** [comment]
I was initially having trouble replicating this issue--turns out I'd set my OS system preferences to always show scroll bars.

So if you'd like to fix your own user experience with the Privacy Badger plugin (and Chrome in general):
- Open "System Preferences"
- Select "General"
- Select the "Show Scroll Bars" -> "Always" radio button

That being said, I'll still have a look into fixing this for everyone else...

**Comment by Zakkai** [comment]
Ahh, good to know.


Z.

**Comment by SwartzCr** [comment]
this seems like a pretty big problem, there's no way we can override this in chrome or ff settings on macos?

**Comment by Zakkai** [comment]
I feel like putting a border around the scrollable region would be good
enough, and wouldn't be confusing if you had scroll bars turned on.


Z.

**Comment by cooperq** [comment]
we could look into a setting, but it seems unlikely so far since it is a part of the OS chrome. Maybe a border region would work, but I'm not convinced of that. I think we should figure out if there is a setting we can tweak first of all. If there is not, I think we should consider and do some testing about what might be the best way to indicate that there is a region below. I wonder if we could trick the scrollbar into appearing by jittering the window?

**Comment by karisafari** [comment]
The challenge with “jittering” is that if a Mac user has their preferences set to show scrollbars only “when scrolling” the scrollbar will disappear again if the jittering stops.

I’ve got a fix working locally by using the [::-webkit-scrollbar pseudo-element] (https://www.webkit.org/blog/363/styling-scrollbars/) to define a custom-styled scrollbar.  With a custom scrollbar defined, webkit will ignore its built-in scrollbar rendering, and we won’t have to worry about the scrollbar being hidden by default on Macs anymore.

My to-dos:
-	detect OS so that this is only applied to Mac
-	make custom scrollbar less ugly

Sound good?

![image](https://cloud.githubusercontent.com/assets/5612143/9952887/e4b3188a-5d8f-11e5-87fa-7d6a031514a3.png)

**Comment by cooperq** [comment]
This looks like a perfect solution! An open question is whether firefox has the same issue, and if so can we do the same thing (or similar) there?
